src/sdk/pynni/nni/compression/torch/pruners.py

```diff
@@ -202,7 +202,7 @@ def __init__(self, model, config_list):
         self.global_threshold = torch.topk(all_bn_weights.view(-1), k, largest=False)[0].max()
         self.register_buffer("if_calculated", torch.tensor(0)) # pylint: disable=not-callable
```

Comment:
One of the benefits to call calc_mask on step instead of forward is that we do not need to register buffer for flag variables like `if_calculated`, here we can switch `if_calculated` to original version to use a dict or set as a member of pruner.

Reply:
Then these variable won't be saved by `torch.state_dict() `?

Reply:
Do you mean to save `if_calculated` as a part of model's state_dict?
The `if_calculated` is an pruner's internal variable during pruning process, we could have a lot of this kind of variable, such as index of epoch, index of batch, etc.  We should not save these.

Reply:
If for resume, the entire pruner should be saved and `if_calcuated` is a part of pruner, we can consider this later when it is necessary.

Reply:
In new version pruner register_buffer API, only buffer with type torch.Tensor will be registered as buffer, otherwise it will be a regular variable in module. Though the name of function is misleading, we can work on a better name.

Reply:
maybe `set_wrappers_attribute` ?

Reply:
Done.